modules/caddyhttp/headers/caddyfile.go

```diff
@@ -149,13 +163,36 @@ func parseReqHdrCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler,
 // complete the substring replacement; in that case, any + or -
 // prefix to field will be ignored.
 func CaddyfileHeaderOp(ops *HeaderOps, field, value, replacement string) {
+	applyCaddyfileHeaderOp(ops, field, value, replacement, nil)
+}
+
+func applyCaddyfileHeaderOp(ops *HeaderOps, field, value, replacement string, respHeaderOps *RespHeaderOps) {
 	if strings.HasPrefix(field, "+") {
 		if ops.Add == nil {
 			ops.Add = make(http.Header)
 		}
 		ops.Add.Set(field[1:], value)
 	} else if strings.HasPrefix(field, "-") {
 		ops.Delete = append(ops.Delete, field[1:])
+	} else if strings.HasPrefix(field, "?") {
+		if respHeaderOps == nil {
+			// "?" prefix only makes sense for response headers.
+			return
```

Comment:
Hmm, I find it odd that it'll silently ignore here. Would be nicer if it could bubble up an error instead, I think. Maybe too big a refactor though. :man_shrugging:

Reply:
It is true that it can be misleading. I made the necessary refactoring in 95aa9ab59429789a5c44b080998c1d0340ed8544 . Trying to use `?` in the wrong context would produce an error similar to this:

```
parsing caddyfile tokens for 'reverse_proxy': Caddyfile:8 - Error during parsing: ?Foo: this header operation cannot be used here
```